### PR TITLE
Remove fork

### PR DESCRIPTION
**Pull request by ktbarrett**
It's happening...

### CODE REVIEW

File: documentation/source/coroutines.rst
Comment (cmarqu) [inline documentation/source/coroutines.rst:144]: Isn't this rewriting history?
Maybe better as
```suggestion
    including :keyword:`yield` statements and ``cocotb.fork`` (since replaced with :func:`~cocotb.start` and :func:`~cocotb.start_soon`).
```
(if at all).

File: cocotb/decorators.py
Comment (marlonjames) [inline cocotb/decorators.py:49]: `_Task()` doesn't take a `parent` object.

**Review by cmarqu — APPROVED** [review]

**Comment by ktbarrett** [comment]
I definitely want more of a consensus on this, any opinion @marlonjames @themperek @imphil?

**Review by marlonjames — APPROVED** [review]
It's missing a news fragment, but otherwise looks good.

It's nice to start removing deprecated things.

File: documentation/source/coroutines.rst
Comment (imphil) [inline documentation/source/coroutines.rst:67]:
```suggestion
    In cocotb 1.x, ``cocotb.fork`` was typically used to run coroutines concurrently,
```

Relative time references don't age well in documentation, let's be specific.

File: documentation/source/coroutines.rst
Comment (imphil) [inline documentation/source/coroutines.rst:150]: I guess we'll need to have a deeper discussion about those versionchanged statements, but I could see us removing all 1.x ones going forward, otherwise the documentation turns into a distributed changelog at some point.

I added this point to the maintainers meeting agenda.

**Review by imphil — APPROVED** [review]
LGTM. Let's get #3436 in first, but then I'm happy to get this merged.

File: documentation/source/coroutines.rst
Comment (imphil) [reply documentation/source/coroutines.rst:150]: We didn't get to it in the meeting. Just keep what you did, we can do a global cleanup once we know what we want to do. (And absolutely don't block this PR on it.)

File: documentation/source/coroutines.rst
Comment (ktbarrett) [reply documentation/source/coroutines.rst:67]: Given the other comment, maybe we should just remove this note.

**Pull request merged**
